frontend/src/components/UpdateModal.tsx

```diff
@@ -1,22 +1,32 @@
-import { useState, FunctionComponent, useEffect, useMemo } from "react";
+import { useState, useEffect, useMemo } from "react";
 import { Button } from "react-bootstrap";
 import Modal from "react-bootstrap/Modal";
 import { AgGridReact } from "ag-grid-react";
 import "ag-grid-enterprise";
 import styles from "./records.module.scss";
 import "ag-grid-community/styles/ag-grid.css";
 import "ag-grid-community/styles/ag-theme-alpine.css";
-import { SampleChange, ChangeForSubmit } from "../shared/helpers";
+import { SampleChange } from "../shared/helpers";
 import { Sample, useUpdateSamplesMutation } from "../generated/graphql";
 import _ from "lodash";
 
-export const UpdateModal: FunctionComponent<{
+interface UpdateModalProps {
   changes: SampleChange[];
   onSuccess: () => void;
   onHide: () => void;
   samples: Sample[];
   onOpen?: () => void;
-}> = ({ changes, onHide, onSuccess, onOpen, samples }) => {
+  sampleKeyForUpdate: keyof Sample;
+}
+
+export function UpdateModal({
+  changes,
+  onHide,
+  onSuccess,
+  onOpen,
+  samples,
+  sampleKeyForUpdate,
+}: UpdateModalProps) {
```

Comment:
Minor refactoring to give a name to the definition of `UpdateModal` function component. This is how we're doing it for other components.